pep-0690.rst

```diff
@@ -115,12 +115,16 @@ directly out of a module ``__dict__``, the only way to reliably prevent
 accidental leakage of lazy objects is to have the dictionary itself be
 responsible to ensure resolution of lazy objects on lookup.
 
-To avoid a performance penalty on the vast majority of dictionaries which never
-contain any lazy objects, we set a specialized lookup kind (`DictKeysKind
-<https://github.com/python/cpython/blob/3.11/Include/internal/pycore_dict.h#L80>`_)
-for module namespace dictionaries when they first gain a lazy-object value. When
-a lazy lookup kind is set and lookup finds that the key references a lazy
-object, it resolves the lazy object immediately before returning it.
+When a lookup finds that the key references a lazy object, it resolves the lazy
+object immediately before returning it. To avoid having side effects mutating
+dictionaries midway an iteration, all lazy objects in a dictionary are resolved
+prior starting an iteration; this could incur a performance penalty when using
+bulk iterations (``iter(dict)``, ``reversed(dict)``, ``dict.__reversed__()``,
+``dict.keys()``, ``iter(dict.keys())`` and ``reversed(dict.keys())``). To avoid
+this performance penalty on the vast majority of dictionaries, which never
+contain any lazy objects, we steal a bit from the ``dk_kind`` field
+(`dk_lazy_imports` <https://github.com/python/cpython/blob/3.11/Include/internal/pycore_dict.h#L105>`_),
```

Comment:
This link line number seems wrong. It's also confusing to link the text `dk_lazy_imports` to a version of the code (3.11 branch) that doesn't have `dk_lazy_imports` in it. You probably don't need to link to the code at all here?
```suggestion
for a new `dk_lazy_imports` bitfield```